// @supabase/auth-ui-react/dist/cjs/src/components/Auth/interfaces/ForgottenPassword.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _tslib = require('../../../../_virtual/_tslib.js');
var jsxRuntime = require('react/jsx-runtime');
var react = require('react');
var constants = require('../../../constants.js');
var Anchor = require('../../UI/Anchor.js');
var Button = require('../../UI/Button.js');
var Container = require('../../UI/Container.js');
require('../../UI/Divider.js');
var Input = require('../../UI/Input.js');
var Label = require('../../UI/Label.js');
var Message = require('../../UI/Message.js');

function ForgottenPassword(_ref) {
  let {
    setAuthView,
    supabaseClient,
    redirectTo,
    i18n,
    appearance,
    showLinks
  } = _ref;

  var _a;

  const [email, setEmail] = react.useState('');
  const [error, setError] = react.useState('');
  const [message, setMessage] = react.useState('');
  const [loading, setLoading] = react.useState(false);

  const handlePasswordReset = e => _tslib.__awaiter(this, void 0, void 0, function* () {
    e.preventDefault();
    setError('');
    setMessage('');
    setLoading(true);
    const {
      error
    } = yield supabaseClient.auth.resetPasswordForEmail(email, {
      redirectTo
    });
    if (error) setError(error.message);else setMessage('Check your email for the password reset link');
    setLoading(false);
  });

  const labels = i18n === null || i18n === void 0 ? void 0 : i18n.forgotten_password;
  return jsxRuntime.jsx("form", Object.assign({
    id: "auth-forgot-password",
    onSubmit: handlePasswordReset
  }, {
    children: jsxRuntime.jsxs(Container.Container, Object.assign({
      gap: "large",
      direction: "vertical",
      appearance: appearance
    }, {
      children: [jsxRuntime.jsxs(Container.Container, Object.assign({
        gap: "large",
        direction: "vertical",
        appearance: appearance
      }, {
        children: [jsxRuntime.jsxs("div", {
          children: [jsxRuntime.jsx(Label.Label, Object.assign({
            htmlFor: "email",
            appearance: appearance
          }, {
            children: labels === null || labels === void 0 ? void 0 : labels.email_label
          })), jsxRuntime.jsx(Input.Input, {
            name: "email",
            type: "email",
            placeholder: labels === null || labels === void 0 ? void 0 : labels.email_input_placeholder,
            onChange: e => setEmail(e.target.value),
            appearance: appearance
          })]
        }), jsxRuntime.jsx(Button.Button, Object.assign({
          type: "submit",
          color: "primary",
          loading: loading,
          appearance: appearance
        }, {
          children: loading ? labels === null || labels === void 0 ? void 0 : labels.loading_button_label : labels === null || labels === void 0 ? void 0 : labels.button_label
        }))]
      })), showLinks && jsxRuntime.jsx(Anchor.Anchor, Object.assign({
        href: "#auth-sign-in",
        onClick: e => {
          e.preventDefault();
          setAuthView(constants.VIEWS.SIGN_IN);
        },
        appearance: appearance
      }, {
        children: (_a = i18n === null || i18n === void 0 ? void 0 : i18n.sign_in) === null || _a === void 0 ? void 0 : _a.link_text
      })), message && jsxRuntime.jsx(Message.Message, Object.assign({
        appearance: appearance
      }, {
        children: message
      })), error && jsxRuntime.jsx(Message.Message, Object.assign({
        color: "danger",
        appearance: appearance
      }, {
        children: error
      }))]
    }))
  }));
}

exports.ForgottenPassword = ForgottenPassword;
