// @supabase/auth-ui-react/dist/esm/src/components/Auth/interfaces/ForgottenPassword.js
import { __awaiter } from '../../../../_virtual/_tslib.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { useState } from 'react';
import { VIEWS } from '../../../constants.js';
import { Anchor } from '../../UI/Anchor.js';
import { Button } from '../../UI/Button.js';
import { Container } from '../../UI/Container.js';
import '../../UI/Divider.js';
import { Input } from '../../UI/Input.js';
import { Label } from '../../UI/Label.js';
import { Message } from '../../UI/Message.js';

function ForgottenPassword(_ref) {
  let {
    setAuthView,
    supabaseClient,
    redirectTo,
    i18n,
    appearance,
    showLinks
  } = _ref;

  var _a;

  const [email, setEmail] = useState('');
  const [error, setError] = useState('');
  const [message, setMessage] = useState('');
  const [loading, setLoading] = useState(false);

  const handlePasswordReset = e => __awaiter(this, void 0, void 0, function* () {
    e.preventDefault();
    setError('');
    setMessage('');
    setLoading(true);
    const {
      error
    } = yield supabaseClient.auth.resetPasswordForEmail(email, {
      redirectTo
    });
    if (error) setError(error.message);else setMessage('Check your email for the password reset link');
    setLoading(false);
  });

  const labels = i18n === null || i18n === void 0 ? void 0 : i18n.forgotten_password;
  return jsx("form", Object.assign({
    id: "auth-forgot-password",
    onSubmit: handlePasswordReset
  }, {
    children: jsxs(Container, Object.assign({
      gap: "large",
      direction: "vertical",
      appearance: appearance
    }, {
      children: [jsxs(Container, Object.assign({
        gap: "large",
        direction: "vertical",
        appearance: appearance
      }, {
        children: [jsxs("div", {
          children: [jsx(Label, Object.assign({
            htmlFor: "email",
            appearance: appearance
          }, {
            children: labels === null || labels === void 0 ? void 0 : labels.email_label
          })), jsx(Input, {
            name: "email",
            type: "email",
            placeholder: labels === null || labels === void 0 ? void 0 : labels.email_input_placeholder,
            onChange: e => setEmail(e.target.value),
            appearance: appearance
          })]
        }), jsx(Button, Object.assign({
          type: "submit",
          color: "primary",
          loading: loading,
          appearance: appearance
        }, {
          children: loading ? labels === null || labels === void 0 ? void 0 : labels.loading_button_label : labels === null || labels === void 0 ? void 0 : labels.button_label
        }))]
      })), showLinks && jsx(Anchor, Object.assign({
        href: "#auth-sign-in",
        onClick: e => {
          e.preventDefault();
          setAuthView(VIEWS.SIGN_IN);
        },
        appearance: appearance
      }, {
        children: (_a = i18n === null || i18n === void 0 ? void 0 : i18n.sign_in) === null || _a === void 0 ? void 0 : _a.link_text
      })), message && jsx(Message, Object.assign({
        appearance: appearance
      }, {
        children: message
      })), error && jsx(Message, Object.assign({
        color: "danger",
        appearance: appearance
      }, {
        children: error
      }))]
    }))
  }));
}

export { ForgottenPassword };
